test/storage.test.js

```diff
@@ -129,6 +133,62 @@ describe('storage.js', () => {
             assert.strictEqual(fs.readFileSync(source.path).toString(), 'Hello, World!');
             assert.ok(nock.isDone());
         });
+      
+        it('should upload data uri to storage and return presigned url mock when disabled download is true', async () => {
```

Comment:
Is there already a test for... : 
- disable download is false => there should be no behavior change
- error handling: what happens if a presigned URL can't be generated?

Reply:
- Yes disableSourceDownload is false for most of our test cases implicitly, the ones newly added are the ones that has a explicit false
- If there are errors in generating preSignUrl I think we should just log the error and not throw error, as the dependency is there only for downstream services and it works today for the worker without this change today Thoughts? @tmathern

Reply:
We should fail early if we can, to avoid unnecessary calls (avoids network bloating and additional costs that would be "for nothing"). If we can't generate the URL, eventually it would fail since one of the services could not get its parameters... And if the service calls fail, eventually the worker will fail the rendition too. Note that this case should be rare, though.

Reply:
Threw SourceUnSupportedError and did not catch in the caller